core/embed/trezorhal/stm32u5/haptic/drv2625/drv2625.c

```diff
@@ -75,45 +60,66 @@
 #error "Must define either LRA or ERM"
 #endif
 
-#define PRESS_EFFECT_AMPLITUDE 25
-#define PRESS_EFFECT_DURATION 10
+// Driver state
+typedef struct {
+  // Set if driver is initialized
+  bool initialized;
+  // Set if driver is enabled
+  bool enabled;
+  // Set to if real-time playing is activated.
+  // This prevents the repeated set of `DRV2625_REG_MODE` register
+  // which would otherwise stop all playback.
+  bool playing_rtp;
 
-#define MAX_AMPLITUDE 127
-#define PRODTEST_EFFECT_AMPLITUDE 127
+} haptic_driver_t;
 
-bool haptic_enabled = true;
+// Haptic driver instance
+static haptic_driver_t g_haptic_driver;
```

Comment:
General question. Should we explicitly zero-initialize these driver structs? they are implicitly zero-initialized but maybe would be easier to understand

Reply:
That's a question. IMHO, it depends on whether such initialization leads to moving the variable into the `.data` section. If it does, I would rather leave it as it is. If not, it would be better to explicitly initialize it.

Reply:
```c
static haptic_driver_t g_haptic_driver = {
  .initialized = false,
}
```

looks fine

Reply:
It seems that if the initialization results in zeroing the struct, GCC places it in the .bss section. So we can be explicit in these cases. See 0d4c7c481c2ca18e07cf8ea44c67e704f40b6cba

Reply:
yeah that's what i would expect. 

i actually meant explicitly zeroing the whole struct like 
`static haptic_driver_t g_haptic_driver = {0}`

this also works, so feel free to go with a variant you prefer